### ISSUE TITLE
Sidewinder FFB Wheel back-powers adapter - potential for damage?

### ISSUE THREAD

**Issue by ej113** [tloimu/adapt-ffb-joy#49]
The wheel is powered differently to the FFP joystick and is able to back-power the adapter through the signal pins. 

The joystick processor is not powered independently: 

1. When the 12V power cable is connected on its own the green LED at the base of the stick does not light up.
2. When the gameport cable is connected on its own (with power through the adapter) the LED flashes, the adapter is able to connect to the stick and everything works except for the force feedback motors.
3. When the gameport and 12V power cables are connected the LED is steady and the motors also provide feedback.

The wheel processor is powered independently:

1. When the 12V power cable is connected on its own the green LED behind the "FORCE" button flashes. The wheel motor performs an initialization from one side back to the center - presumably to self-calibrate.
2. When the gameport cable is connected on its own (with power through the adapter) there is no response and no LED. The adapter cannot connect to the wheel.
3. When the gameport and 12V power cables are connected the LED and feedback motors can both be toggled on and off with the FORCE button and the adapter can communicate with the wheel. 

However when the adapter USB cable (which provides power to the adapter MCU) is then removed the adapter continues to be powered and the wheel behaves as if it is still connected. This apparently can occur when there is a voltage on the input pin(s) and current is drawn through the internal protection diodes to the VCC rail. 

I guess it's possible my wheel is malfunctioning, but I did also find an [image](https://www.reddit.com/r/simracing/comments/qx3uhn/trying_to_get_a_ms_sidewinder_force_feedback/?rdt=34126) of another wheel LED lit despite the gameport cable disconnected. 

I have measured the voltages on the wheel gameport pins with a multimeter with respect to Ground (Gameport Pin 4).
Pin 1: 0V (VCC)
Pins 2,3,7,10&11: 4.9V (X1,X2 and all buttons)
Pin 12: 4.7V (MIDI out)

Also when back-powered I measured the voltage between adapter GND and VCC as 3.3V

According to the datasheet for the Atmega32u4 under Absolute Maximum Ratings:
` Voltage on any Pin except RESET and VBUS with respect to Ground (8) . . . . . . . . -0.5V to VCC +0.5V`

So this condition has potential to damage the adapter MCU. If the current is sufficient it could be bad for the wheel as well. 

The option to ignore this and try to always ensure the adapter is connected to USB first is risky and has a number of drawbacks:
- It's easy to forget - I've made the mistake already a few times while testing it. It doesn't help that the LED can already be off and the power cable port is well hidden behind the wheel. 
- The ideal initialization sequence appears to be to connect power first then gameport/USB. When the adapter connects too quickly the self-calibration aborts and the wheel is off-centre. Maybe this can be fixed in firmware.
-   To reset the adapter requires two connectors to be unplugged then replugged.

There appear to be a number of alternative solutions using basic components (e.g. resistors, diodes, tri-state buffers) that would all involve modification to the adapter circuitry. Note there are 6 powered pins so the MCU would need to be protected from voltages on all of these. I'm not well enough versed in Electrical Engineering to predict what would be the simplest/cheapest solution and which might also cause issues with communications. Any ideas?

**Comment by Harbor-Gaming** [comment]
Hey there, idk much about electronics, I have a Microsoft sidewinder force feedback wheel with pedals, but it is the gameport version. I wanna connect it to USB. Does this adapter for joystick work aswell for for the wheel? Cuz some pins the wheel has aren't connected to anything in the schematics and vice versa. If you it doesn't work or I need to adapt it, don't give me a complex explanation but dumb it down for me please. Thank you for your attention.

**Comment by beel1** [comment]
Thanks @ej113 for pointing this out, I may have a few ideas:

>According to the datasheet for the Atmega32u4 under Absolute Maximum Ratings:
>```
>Voltage on any Pin except RESET and VBUS with respect to Ground (8) . . . . . . . . -0.5V to VCC +0.5V
>```
>So this condition has potential to damage the adapter MCU.

From the Atmel AVR040 App Note:
```
4.9.1. General I/O Pin Protection
All general I/O-pins have internal ESD protection diodes to GND and VCC [...]. If exceeding pin voltage 'Absolute Maximum Ratings' in the datasheet, resulting currents can harm the device if they are not limited accordingly.
```

So the voltage at the pins may not be dangerous in itself but the current could damage the protection diodes.
The only pins that have no current limitation resistors are the buttons inputs PD0 and PB0 to PB3. To keep the BOM simple, a 220R resistor in series (just like PD3) for each button pin should prevent damage in case something goes very wrong.
I quickly made a PoC and verified that this modification doesn't cause any issue on communication.

>Also when back-powered I measured the voltage between adapter GND and VCC as 3.3V

This also could be a concern, because the 32u4 shouldn't be clocked at 16MHz below 4.5V VCC and may act weird.
To prevent this in software you can set the Brown-Out Detector at 4.5V. Unfortunately the only way to do that is to burn internal fuses which is not possible with the Teensy Loader.
Otherwise in hardware you can use a reset monitoring circuit with a threshold of 4.5V.

>The ideal initialization sequence appears to be to connect power first then gameport/USB. When the adapter connects too quickly the self-calibration aborts and the wheel is off-centre. Maybe this can be fixed in firmware.

Here is how I avoid the off-centered wheel for now:
1. Connect the wheel gameport to the USB adapter
2. **Turn the wheel fully to the left**
3. Connect the adapter to an USB port, it starts blinking
4. Apply power to the wheel, it performs the shortest self-calibration (goes from left to centre then stops)
5. Toggle the "FORCE" button then play :)

But I agree a firmware fix should be possible.
